run_robot_test.sh

```diff
@@ -114,24 +113,25 @@ setup_venv() {
 
 create_artifact_dir() {
     # Create a unique directory to store the output for current test run
-    if [[ ! -d "${TEST_ARTIFACT_DIR}" ]]; then
-      mkdir "${TEST_ARTIFACT_DIR}"
-    fi
     case "$(uname -s)" in
         Darwin)
-            TEST_ARTIFACT_DIR=$(mktemp -d  ${TEST_ARTIFACT_DIR} -t ${TEST_ARTIFACT_DIR}/rhoda-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+            TEST_ARTIFACT_DIR="${TEST_ARTIFACT_DIR}/rhoda-ci-$(date +%Y-%m-%d-%H-%M)-XXXX"
              ;;
         Linux)
-            TEST_ARTIFACT_DIR=$(mktemp -d -p ${TEST_ARTIFACT_DIR} -t rhoda-ci-$(date +%Y-%m-%d-%H-%M)-XXXXXXXXXX)
+            TEST_ARTIFACT_DIR="${TEST_ARTIFACT_DIR}/rhoda-${BUILD_NUMBER}-$(date +%Y-%m-%d-%M%S)"
```

Comment:
${BUILD_NUMBER} I dont see the declaration for this variable, is this expected to be passed as CLI argument?

Reply:
${BUILD_NUMBER} is a Jenkins Environment variable, it will automatically get picked up once a Job is kicked off
https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables
When running manually, the variable will be empty, so won't be included in the directory name

Reply:
I am not sure Jenkins environment variables can be called this way, I use ${env.BUILD_NUMBER} to access the variable.
Can you please attach the test results from local jenkins setup?